Can't index an 2d array like `a1[int, tuple]`

I tried to use pythran to improve the following function (it is largely simplified for better debugging):

```
import numpy as np


#pythran export calc_bin_more_pythran2(int, int[:], int[:])
def calc_bin_more_pythran2(Vdim, binnumbers, nbin):
    flatcount = np.bincount(binnumbers, None)
    a = np.nonzero(flatcount)

    result = np.empty([Vdim, nbin.prod()], float)
    result[0, a] = 1.0/flatcount[a]
    return result
```

And I got this error. This error message is wrong because I've tested that `np.bincount(binnumbers, None)` is right and can be compiled successfully.
```
(scipy-dev) charlotte@CHARLOTLIU-MB0 stats % pythran test_bincount.py
WARNING: Compilation error, trying hard to find its origin...
CRITICAL: You shall not pass!
E: test_bincount.py:6:16 error: Invalid argument type for function call to `Callable[[T0, none], ...]`, no overload found, tried:
Callable[[Iterable[bool], Iterable[float], int], Array[1d, int]]
Callable[[Iterable[bool], Iterable[float]], Array[1d, int]]
Callable[[Iterable[bool]], Array[1d, int]]
Callable[[Iterable[int], Iterable[float], int], Array[1d, int]]
Callable[[Iterable[int], Iterable[float]], Array[1d, int]]
Callable[[Iterable[int]], Array[1d, int]]
----
    flatcount = np.bincount(binnumbers, None)
                ^~~~ (o_0)
----
```

After much trying, I found it is related to 2d array indexing because if I modify `result` to a 1d array `b`, it works just fine.
```
import numpy as np


#pythran export calc_bin_more_pythran2(int, int[:], int[:])
def calc_bin_more_pythran2(Vdim, binnumbers, nbin):
    flatcount = np.bincount(binnumbers, None)
    a = np.nonzero(flatcount)

    b = np.empty([nbin.prod()], float)
    b[a] = 1.0/flatcount[a]
    return b
```

cc @rgommers

Thanks! That works. 

However, back to the original function, pythran seems fail on `flatsum = np.bincount(binnumbers, values[vv])`. If I delete `flatsum` in `return result, flatsum`, it works. If not, it will fail. The error message points to ` flatcount = np.bincount(binnumbers, None)` but it is actually good.

```
(scipy-dev) charlotte@CHARLOTLIU-MB0 stats % pythran test_bincount.py
WARNING: Compilation error, trying hard to find its origin...
CRITICAL: You shall not pass!
E: test_bincount.py:6:16 error: Invalid argument type for function call to `Callable[[T0, none], ...]`, no overload found, tried:
Callable[[Iterable[bool], Iterable[float], int], Array[1d, int]]
Callable[[Iterable[bool], Iterable[float]], Array[1d, int]]
Callable[[Iterable[bool]], Array[1d, int]]
Callable[[Iterable[int], Iterable[float], int], Array[1d, int]]
Callable[[Iterable[int], Iterable[float]], Array[1d, int]]
Callable[[Iterable[int]], Array[1d, int]]
----
    flatcount = np.bincount(binnumbers, None)
                ^~~~ (o_0)
----
```

This is my code:
```
import numpy as np
import builtins

#pythran export calc_bin_more_pythran2(int, int[:], int[:], float[:,:])
def calc_bin_more_pythran2(Vdim, binnumbers, nbin, values):
    flatcount = np.bincount(binnumbers, None)
    a = flatcount.nonzero()
    result = np.empty([Vdim, nbin.prod()], float)
    for vv in builtins.range(Vdim):
        flatsum = np.bincount(binnumbers, values[vv])
        for i in a:
            result[vv, i] = 1 / flatcount[i]
            # result[vv, i] = flatsum[i] / flatcount[i]

    return result, flatsum
```

--- a/pythran/pythonic/numpy/bincount.hpp
+++ b/pythran/pythonic/numpy/bincount.hpp
@@ -46,7 +46,7 @@ namespace numpy
                                 std::declval<typename E::dtype>()),
                        types::pshape<long>>>::type
     out(types::pshape<long>(length), 0L);
-    auto iweight = weights.fbegin();
+    auto iweight = weights.begin();
     for (auto iter = expr.fbegin(), end = expr.fend(); iter != end;
          ++iter, ++iweight)
       out[*iter] += *iweight;

--- a/pythran/tests/test_numpy_func2.py
+++ b/pythran/tests/test_numpy_func2.py
@@ -385,6 +385,9 @@ def test_copy0(x):
     def test_bincount2(self):
         self.run_test("def np_bincount2(a, w): from numpy import bincount; return bincount(a + 1,w)", numpy.array([0, 1, 1, 2, 2, 2]), numpy.array([0.3, 0.5, 0.2, 0.7, 1., -0.6]), np_bincount2=[NDArray[int,:], NDArray[float,:]])
 
+    def test_bincount3(self):
+        self.run_test("def np_bincount3(a): from numpy import bincount,ones; x=ones((a.shape[0]+2), dtype=int); return bincount(a, x[:-2])", numpy.array([0, 1, 1, 2, 2, 2]), np_bincount3=[NDArray[int,:]])
+
     def test_binary_repr0(self):
         self.run_test("def np_binary_repr0(a): from numpy import binary_repr ; return binary_repr(a)", 3, np_binary_repr0=[int])
 

